packages/terra-show-hide/src/ShowHide.jsx

```diff
@@ -55,29 +55,22 @@ const defaultProps = {
 };
 
 /**
- * Remove attributes that makes element focusable.
- */
-const removeAttributes = (event) => {
-  event.currentTarget.removeAttribute('tabindex');
-  event.currentTarget.removeAttribute('role');
-};
-
-/**
- * Focus element.
- */
-const focusElement = (element) => {
-  // try to focus the element.
-  element.focus();
-  // if element is not focusable, add atributes that make it focusable, then focus again.
-  if (document.activeElement !== element) {
-    element.setAttribute('tabIndex', '-1');
-    element.setAttribute('role', 'group');
-    element.addEventListener('blur', (event) => {
-      removeAttributes(event);
-    });
-    element.focus();
-  }
-};
+  * Returns true if element is NOT normally interactive element like button and has no negative tabindex added by user.
+  * Considers following options:
+  *
+  * (1) Focusable BUTTON element would have:
+  * element.tabIndex:  0
+  * element.hasAttribute():  false
+  *
+  * (2) Not focusable SPAN element with no tabindex set would have:
+  * element.tabIndex:  -1
+  * element.hasAttribute():  false
+  *
+  * (3) Focusable SPAN element with tabindex set to -1 would have:
+  * element.tabIndex:  -1
+  * element.hasAttribute():  true
+  */
+const checkIfAddTabindex = (element) => element?.tabIndex === -1 && !element?.hasAttribute('tabindex');
```

Comment:
We would only need to do this if the focusRef is passed in.  Can't we create a const value that lets us know whether to update the tabIndex.  The check should only be needed once?

Reply:
Correct, that is how it works now, as I call that method only one time, if the focusRef is passed in: https://github.com/cerner/terra-core/blob/6d1d09062f40aaba64069bfac4eba48e6d2d6dfe/packages/terra-show-hide/src/ShowHide.jsx#L99-L100